### DISCUSSION TITLE
Does OpenSTA support displaying the corresponding error code when an error occurs?

### DISCUSSION THREAD

**Discussion by Zhihui-SJTU** [parallaxsw/OpenSTA#112] [Q&A] [answered]
For example, in Figure 1, when I was running my own TCL script in OpenSTA, it showed an error. 
![image1](https://github.com/user-attachments/assets/e7a03e1e-f077-4fa8-8236-b7522833fea5)

I searched the error message in the OpenSTA source code and found many similar ones, but they can be distinguished by error codes (as shown in Figure 2). If the error code can be displayed when an error occurs, the user will be able to accurately locate which function and line the error message is coming from.
![image2](https://github.com/user-attachments/assets/ec5db73c-9046-4d00-b1b2-b2ac3582bd96)

**Comment by parallaxsw** [comment, marked as answer]
No such option exists. It is trivial to modify the reporting functions to show the error code but I haven't really seen the need. Openroad prints them.
